website/server/controllers/api-v3/auth.js

```diff
@@ -584,8 +608,14 @@ api.updateEmail = {
 
     if (emailAlreadyInUse) throw new NotAuthorized(res.t('cannotFulfillReq'));
 
-    let candidatePassword = passwordUtils.encrypt(req.body.password, user.auth.local.salt);
-    if (candidatePassword !== user.auth.local.hashed_password) throw new NotAuthorized(res.t('wrongPassword'));
+    let password = req.body.password;
```

Comment:
Suggestion: pull this logic into a `verifyPassword` helper method that checks the password, throws an error if the password is wrong, and converts to the new encryption schema if needed.